metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -276,76 +273,89 @@ final class BloopServers(
     val result =
-      for { // if the requestedBloopJvmProperties and bloopGlobalJsonFilePath are defined
-        requestedBloopJvmProperties <- maybeRequestedBloopJvmProperties
+      for { // if  bloopGlobalJsonFilePath is defined
         bloopGlobalJsonFilePath <- getBloopFilePath(fileName = "bloop.json")
         bloopCreatedByMetalsFilePath <- getBloopFilePath(fileName =
           "created_by_metals.lock"
         )
-      } yield
-        if (
-          maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties &&
-          !(requestedBloopJvmProperties.isEmpty && maybeRunningBloopJvmProperties.isEmpty)
-        ) { // the properties are updated
-          if (
-            bloopGlobalJsonFilePath.exists &&
-            getBloopGlobalJsonLastModifiedTime(
-              bloopGlobalJsonFilePath
-            ) > getBloopGlobalJsonLastModifiedByMetalsTime(
-              bloopCreatedByMetalsFilePath
-            )
-          ) {
-            // the global json file was previously modified by the user through other means;
-            // therefore overwriting it requires user input
-            languageClient
-              .showMessageRequest(
-                Messages.BloopGlobalJsonFilePremodified.params()
-              )
-              .asScala
-              .flatMap {
-                processUserPreferenceForBloopJvmProperties(
-                  _,
-                  bloopGlobalJsonFilePath,
-                  bloopCreatedByMetalsFilePath,
-                  requestedBloopJvmProperties,
-                  maybeJavaHome,
-                  reconnect
-                ) andThen {
-                  case Failure(exception) =>
-                    languageClient.showMessage(
-                      MessageType.Error,
-                      exception.getMessage
-                    )
-                  case Success(_) => Future.successful()
-                }
-              }
-          } else {
-            // bloop global json file did not exist; or it was last modified by metals;
-            // hence it can get created or overwritten by Metals with no worries
-            // about overriding the user preferred settings
-            updateBloopGlobalJsonFileThenRestart(
-              bloopGlobalJsonFilePath,
-              bloopCreatedByMetalsFilePath,
-              requestedBloopJvmProperties,
-              maybeJavaHome,
-              reconnect
-            ) andThen {
-              case Failure(exception) =>
-                languageClient.showMessage(
-                  MessageType.Error,
-                  exception.getMessage
-                )
-              case Success(_) => Future.successful()
-            }
-          }
-        } else Future.successful(())
 
-    result.getOrElse(Future.successful(()))
+        requestedBloopJvmProperties = maybeRequestedBloopJvmProperties
+          .getOrElse(List.empty)
+        if maybeRequestedBloopJvmProperties != maybeRunningBloopJvmProperties ||
```

Comment:
Could we rework `maybeRunningBloopJvmProperties` to read from the `bloop.json` file so that even if we have `None` then we will not show the message about restarting if it's not needed? Currently this shows up every time we start the server.

The same with `maybeRunningMetalsJavaHome`, it will be None in the default value and updated by the client each time. We should check if it's already set as Java home. I am thinking we might need an additional workflow for this actually.

We could set the `bloop.json` at the start of the server before starting Bloop (you can take a look at the BloopServers class) so that in most cases users don't need to do anything and not even ask about anything unless there is a `bloop.json` file already with a different home. In case bloop.json exists with a different home we can show a message to change the javaHome to avoid issues.

Reply:
@tgodzik not sure if I understood what should be done for javaHome! But with the changes I just pushed, Metals does not propose to restart Bloop anymore, if the user requested setting matches the contents of the global json file! You can take a look!